documentation/general/torn-sdk.md

```diff
@@ -0,0 +1,109 @@
+# Torn .NET SDK
+
+## Terminology
+
+- msbuild: refers the .NET Framework based msbuild included with Visual Studio.
+- dotnet msbuild: refers to the .NET Core based msbuild included with the .NET SDK.
+- analyzers: this document will use analyzers to refer to both analyzers and generators.
+
+## Summary
+
+Visual Studio and .NET SDK are separate products but they are intertwined in command line and design time build scenarios as different components are loaded from each product.
+
+| Scenario | Loads Roslyn | Loads Analyzers / Generators |
+| --- | --- | --- |
+| msbuild | From Visual Studio | From .NET SDK |
+| dotnet msbuild | From .NET SDK | From .NET SDK |
+| Visual Studio Design Time | From Visual Studio | From Both |
+
+Generally this mixing of components is fine because Visual Studio will install a .NET SDK that is functionally paired with it. That is the compiler and analyzers are the same hence mixing poses no real issue. For example 17.10 installs .NET SDK 8.0.3xx, 17.9 installs .NET SDK 8.0.2xx, etc ... However when the .NET SDK is not paired with the Visual Studio version,then compatibility issues can, and will, arise. Particularly when the .NET SDK is _newer_ than Visual Studio customers will end up with the following style of error:
+
+> CSC : warning CS9057: The analyzer assembly '..\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk.Razor\source-
+generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll' references version '4.9.0.0' of the compiler, which is newer
+than the currently running version '4.8.0.0'.
+
+To address this issue we are going to separate Visual Studio and the .NET SDK in build scenarios. That will change our matrix to the following:
+
+| Scenario | Loads Roslyn | Loads Analyzers / Generators |
+| --- | --- | --- |
+| msbuild | From .NET SDK | From .NET SDK |
+| dotnet msbuild | From .NET SDK | From .NET SDK |
+| Visual Studio Design Time | From Visual Studio | From Visual Studio |
+
+Specifically we will be
+
+1. Changing msbuild to use a compiler from the .NET SDK
+2. Changing Visual Studio to use analyzers from Visual Studio
+
+In addition to making our builds more reliable this will also massively simplify our analyzer development strategy. Analyzers following this model can always target the latest Roslyn version without the need for complicated multi-targeting.
```

Comment:
> Analyzers following this model can always target the latest Roslyn version

Do all in-box analyzers _need_ to target latest? Or is this a thing that we would like do to simplify the development/build/etc matrix? I'm wondering if there's some set of analyzers that could just target Ye Olde Roslyn Versions back enough to sidestep the issue.

Reply:
This is a classic it depends; some analyzers won't care and could target Roslyn 1.0. But many will need to update their logic for newer language constructs and can't do that today, so they do things like this: https://github.com/dotnet/roslyn-analyzers/tree/main/src/Utilities/Compiler/Lightup.

Reply:
I will try and clarify a bit better about inbox analyzers. But essentially inbox analyzers have a few choices: 

1. Dual insert into VS: this is the recommended approach. 
2. Reference the oldest supported Roslyn version: this is just fine if it works for you. 
3. Multi-target like NuGet packages: this is not ideal because it increases .NET SDK size and it means that builds are significantly different between `msbuild` and `dotnet build`.